proposals/0008-swift-prometheus.md

```diff
@@ -0,0 +1,155 @@
+# SwiftPrometheus - Prometheus Metrics in Swift
+
+* Proposal: SSWG-0008
+* Authors: [Jari Koopman / LotU](https://github.com/MrLotU)
+* Review Manager: TBD
+* Status:  [Implemented ](https://github.com/MrLotU/SwiftPrometheus/)
+* Pitch: [Pitches/Prometheus](https://forums.swift.org/t/client-side-prometheus-implementation/18098/)
+
+## Package Description
+Prometheus client side implementation.
+
+|  |  |
+|--|--|
+| **Package Name** | `SwiftPrometheus` |
+| **Module Name** | `Prometheus` |
+| **Proposed Maturity Level** | [Sandbox](https://github.com/swift-server/sswg/blob/master/process/incubation.md#process-diagram) |
+| **License** | [Apache 2.0](https://github.com/MrLotU/SwiftPrometheus/blob/master/LICENSE) |
+| **Dependencies** | swift-nio 1.x.x or 2.x.x<sup>[1](#footnote_1)</sup> - swift-metrics > 1.0.0 |
+
+## Introduction
+
+_For a background on [`swift-metrics`](https://github.com/apple/swift-metrics) see the metrics [proposal discussion](https://forums.swift.org/t/discussion-server-metrics-api/19600) and [feedback](https://forums.swift.org/t/feedback-server-metrics-api/21353) threads._
+
+[Prometheus](https://prometheus.io) is a service that scrapes metrics from other services, and stores them in a time series database to be queried and alerted upon. SwiftPrometheus is a client-side library to allow packages to export metrics in the [Prometheus format](https://openmetrics.io) in Swift, with the ability to use it both connected to & separately from swift-metrics.
+
+## Motivation
+
+With Prometheus being one of the most widely used metric reporting tools, as well as a [graduated project of the Cloud Native Computing Foundation](https://www.cncf.io/announcement/2018/08/09/prometheus-graduates/), it's a cornerstone that can not be left out in a serverside ecosystem. This package is created for everyone to use & build upon for their metric reporting.
+
+## Detailed design
+
+SwiftPrometheus works around one base class `PrometheusClient` and some metric types around it. The prometheus metric types are:
+(from the [prometheus docs](https://www.prometheus.io/docs/concepts/metric_types/))
```

Comment:
```suggestion
The prometheus metric types are (from the [prometheus docs](https://www.prometheus.io/docs/concepts/metric_types/)):
```

Reply:
This is on the line above, right?

Reply:
Wait. This just looked different in preview. Nevermind me.